Functions.Templates/Bindings/bindings.json

```diff
@@ -1747,6 +1747,639 @@
                     "help": "$bot_out_secret_help"
                 }
             ]
-        }
+        },
+        {
+            "type": "token",
+            "displayName": "$token_displayName",
+            "direction": "in",
+            "enabledInTryMode": false,
+            "documentation": "$content=Documentation\\Token.md",
+            "filters": ["MSGraph"],
+            "settings": [
+                {
+                    "name": "name",
+                    "value": "string",
+                    "defaultValue": "token",
+                    "required": true,
+                    "label": "$token_in_name_label",
+                    "help": "$token_in_name_help"
+                },
+                {
+                    "name": "Resource",
+                    "value": "string",
+                    "required": false,
+                    "label": "$token_in_resource_label",
+                    "help": "$token_in_resource_help"
+                },
+                {
+                    "name": "PrincipalId",
+                    "value": "appSetting",
```

Comment:
We already have a resource type appSetting in binding.json. Since your control does the same thing, could you merge both. Essentially use your control for "resource": "AppSetting". 
https://github.com/Azure/azure-webjobs-sdk-templates/blob/dev/Functions.Templates/Bindings/bindings.json#L1610